sdk/storage/azure-storage-blobs/src/block_blob_client.cpp

```diff
@@ -115,6 +115,16 @@ namespace Azure { namespace Storage { namespace Blobs {
     constexpr int64_t MaxBlockNumber = 50000;
     constexpr int64_t BlockGrainSize = 1 * 1024 * 1024;
 
+    if (bufferSize <= static_cast<std::size_t>(options.TransferOptions.SingleUploadThreshold))
```

Comment:
We should add a comment here to explain how this fixes the issue and why the check and return needs to be here rather than below.

Reply:
Add a changelog entry please.

Reply:
We don't need a CL entry. The bug was introduced after latest release.